examples/list_box_model/main.rs

```diff
@@ -0,0 +1,190 @@
+                .flags(glib::BindingFlags::DEFAULT | glib::BindingFlags::SYNC_CREATE)
+                .build();
+            hbox.append(&label);
+
+            let spin_button = gtk::SpinButton::with_range(0.0, 100.0, 1.0);
+            item.bind_property("count", &spin_button, "value")
+                .flags(glib::BindingFlags::DEFAULT | glib::BindingFlags::SYNC_CREATE | glib::BindingFlags::BIDIRECTIONAL)
+                .build();
+            hbox.append(&spin_button);
+
+            // When the edit button is clicked, a new modal dialog is created for editing
+            // the corresponding row
+            let edit_button = gtk::Button::with_label("Edit");
+            edit_button.connect_clicked(clone!(@weak window, @strong item => move |_| {
+            let dialog = gtk::Dialog::with_buttons(Some("Edit Item"), Some(&window), gtk::DialogFlags::MODAL,
+                &[("Close", ResponseType::Close)]);
+            dialog.set_default_response(ResponseType::Close);
+            dialog.connect_response(|dialog, _| dialog.close());
+
+            let content_area = dialog.content_area();
+
+            // Similarly to the label and spin button inside the listbox, the text entry
+            // and spin button in the edit dialog are connected via property bindings to
+            // the item. Any changes will be immediately reflected inside the item and
+            // by the listbox
+            let entry = gtk::Entry::new();
+            item.bind_property("name", &entry, "text")
+                .flags(glib::BindingFlags::DEFAULT | glib::BindingFlags::SYNC_CREATE | glib::BindingFlags::BIDIRECTIONAL)
+                .build();
+
+            // Activating the entry (enter) will send response `ResponseType::Close` to the dialog
+            entry.connect_activate(clone!(@weak dialog => move |_| {
+                dialog.response(ResponseType::Close);
+            }));
+            content_area.append(&entry);
+
+            let spin_button = gtk::SpinButton::with_range(0.0, 100.0, 1.0);
+            item.bind_property("count", &spin_button, "value")
+                .flags(glib::BindingFlags::DEFAULT | glib::BindingFlags::SYNC_CREATE | glib::BindingFlags::BIDIRECTIONAL)
+                .build();
+            content_area.append(&spin_button);
+
+            dialog.show();
+        }));
+        hbox.append(&edit_button);
+
+        box_.set_child(Some(&hbox));
+
+        // When a row is activated (select + enter) we simply emit the clicked
+        // signal on the corresponding edit button to open the edit dialog
+        box_.connect_activate(clone!(@weak edit_button => move |_| {
+            edit_button.emit_clicked();
+        }));
+
+        box_.show();
+
+        box_.upcast::<gtk::Widget>()
+    }));
+
+    let scrolled_window = gtk::ScrolledWindow::builder()
+        .hscrollbar_policy(gtk::PolicyType::Never) // Disable horizontal scrolling
+        .min_content_height(480)
```

Comment:
Is this an ok thing to do? As far as I can tell the gtk3 didn't do anything of the sort and the Widget filled the window anyway, but that didn't happen in gtk4

Reply:
Try setting valign to fill in the listbox?

Reply:
That didn't seem to do it in my testing